source/tls_endpoint.go

```diff
@@ -36,7 +37,10 @@ func NewTLSEndpoint(host string, port string) *TLSEndpoint {
 
 // GetCertificates tries to get certificates from endpoint using tls.Dial
 func (e *TLSEndpoint) GetCertificates() ([]*x509.Certificate, error) {
-	conn, err := tls.Dial("tcp", e.Hostname+":"+e.Port, &defaultTLSConfig)
+
+	// We cannot connect to Hostnames with wildcards, so replacing with www.
+	hostName := strings.Replace(e.Hostname, "*", "www", -1)
```

Comment:
`www` is very common sub-domain. And there are high chances to have different `A` record for this and hosted somewhere else (other than Ingress). This way controller will hit some other host to and get that TLS certificate from that host (Not Ingress Certificate) creating falsey results. I would recommend to use better alternative which are rarely used. May be using `wildcard` work better.